cohortextractor2/query_engines/spark_dialect.py

```diff
@@ -128,6 +170,48 @@ def _get_table_columns(self, connection, table_name, schema):
             else:
                 raise
 
+    # We do not currently use this function in our code, but we might in future.
```

Comment:
What might we use it for? Why aren't we using it now? 

Is there an alternative to leaving in code which will probably rot anyway? A comment referring the reader to the _partition information_ weirdness & a historic commit?

Reply:
So, this is a copy of some databricks specific support from the only pre-exisiting sqlalchemy dialect I could find:

https://github.com/crflynn/databricks-dbapi/blob/master/databricks_dbapi/sqlalchemy_dialects/base.py

We don't want to use that module, as we want to use the recommended `databricks-sql-connector` DBAPI not pyhive's DBAPI. And the *only* other thing it brings in is this function. It is used by Sqlalchemy's machinery, but not machinery that we are using currently, but we may in future.

I didn't want to lose the knowledge that this is a databricks/OSS spark difference that could bite us in future, so I included the function. It's supports both OSS Spark and Databricks, so I saw know harm, and it might Just Work when we need it to and prevent an issue.

I'am +0 on keeping it in. If you would prefer to remove it, how can we capture the knowledge somewhere that this might be an issue in future, and if so, here's a fix?

Reply:
Maybe another solutions is getting the change upstreamed into pyhive? It's a ~10 character change, but no idea about testing.

https://github.com/dropbox/PyHive/blob/master/pyhive/sqlalchemy_hive.py#L306